clippy_lints/src/needless_late_init.rs

```diff
@@ -73,6 +77,31 @@ fn contains_assign_expr<'tcx>(cx: &LateContext<'tcx>, stmt: &'tcx Stmt<'tcx>) ->
     seen
 }
 
+fn contains_let(cond: &Expr<'_>) -> bool {
```

Comment:
Is it possible to replace this with `higher::IfLet::hir`?

Reply:
It wouldn't be for let chains